src/forward/lib.jl

```diff
@@ -62,14 +62,20 @@ if VERSION >= v"1.4.0-DEV.304"
     _pushforward((first(args), tail(tail(dargs))...), Core._apply, f, args...)
 end
 
-using ..Zygote: literal_getproperty, literal_getindex
+using ..Zygote: literal_getproperty, literal_getfield, literal_getindex
 
-_pushforward(dargs, ::typeof(getproperty), x, f) =
-  _pushforward(dargs, literal_getproperty, x, Val(f))
+_pushforward(dargs, ::typeof(literal_getproperty), x::NamedTuple, ::Val{f}) where {f} =
```

Comment:
Is `f` some arbitrary set of indices in this context?

Reply:
Presumably it should always be a key from the NT

Reply:
For `getproperty` `f` can only be a symbol. The second argument of `getfield` can also be an integer for `NamedTuples` though.

Reply:
Hmm okay. Is there a more descriptive naming convention that we could adopt other than `f`? If something is called `f`, my immediate reaction is to assume it represents a function of some kind. (I realise I'm nit picking here)

Reply:
> For getproperty f can only be a symbol. The second argument of getfield can also be an integer for NamedTuples though.

This is not true.
`getproperty` can also be a integer on tuples
```
julia> x = (10,20,30)
(10, 20, 30)

julia> getproperty(x, 2)
20
```

Infact you can go weirder if you want
```
julia> struct Foo end

julia> Base.getproperty(::Foo, i::String) = i^3

julia> Foo()."a b c"
"a b ca b ca b c"
```

Reply:
Sorry, I only meant in the case of `NamedTuples`

Reply:
We could call it `property_name`?

Reply:
That would work for me.

Reply:
Ok, should be addressed now